mlir/lib/Dialect/Tensor/Transforms/FoldIntoPackAndUnpackPatterns.cpp

```diff
@@ -21,6 +21,57 @@ static bool areAllConstantIntValue(ArrayRef<OpFoldResult> ofrs, int64_t value) {
       ofrs, [&](OpFoldResult ofr) { return isConstantIntValue(ofr, value); });
 }
 
+/// Helper function to generate an equivalent permutation map for
+/// `linalg.transpose` and `tensor.pack` which will be used after their folding
+/// into a `tensor.pack`.
+static bool getRemappedPermutationForTransposeAndPack(
```

Comment:
I would like to understand a bit better why we have such complexity for `FoldConsumerPackWithProducerLinalgTransposeOp`; in such cases, to me, it is a simple `applyPermutation` on top of the transpose permutation based on outer dims perm, inner tiles and tiles will remain the same.

Reply:
Yes you are correct. 
I disregarded that while working on this PR. I have updated the diff to incorporate this.

Reply:
Sorry I may have confused you with this comment. What I mentioned above is true as long as the transpose do not rearrange the loop ids in `inner_dims_pos`.